src/sentry/static/sentry/app/views/onboarding/wizardNew.jsx

```diff
@@ -0,0 +1,289 @@
+import space from 'app/styles/space';
+import withOrganization from 'app/utils/withOrganization';
+import withProjects from 'app/utils/withProjects';
+
+const recordAnalyticStepComplete = ({organization, project, step}) =>
+  analytics('onboarding_v2.step_compete', {
+    org_id: parseInt(organization.id, 10),
+    project: project ? project.slug : null,
+    step: step.id,
+  });
+
+const ONBOARDING_STEPS = [
+  {
+    id: 'welcome',
+    title: t('Welcome to Sentry'),
+    Component: OnboardingWelcome,
+  },
+  {
+    id: 'select-platform',
+    title: t('Select a platform'),
+    Component: OnboardingPlatform,
+  },
+  {
+    id: 'get-started',
+    title: t('Install the Sentry SDK'),
+    Component: OnboardingProjectSetup,
+  },
+];
+
+class OnboardingWizard extends React.Component {
+  static propTypes = {
+    projects: PropTypes.arrayOf(SentryTypes.Project),
+    organization: SentryTypes.Organization,
+  };
+
+  state = {};
+
+  get activeStepIndex() {
+    return ONBOARDING_STEPS.findIndex(({id}) => this.props.params.step === id);
+  }
+
+  get activeStep() {
+    return ONBOARDING_STEPS[this.activeStepIndex];
+  }
+
+  get firstProject() {
+    const sortedProjects = this.props.projects.sort(
+      (a, b) => new Date(b.dateCreated) - new Date(a.dateCreated)
+    );
+
+    return sortedProjects.length > 0 ? sortedProjects[0] : null;
+  }
+
+  get projectPlatform() {
+    return this.state.platform || this.firstProject?.platform;
+  }
+
+  handleUpdate = data => {
+    this.setState(data);
+  };
+
+  handleNextStep(step, data) {
+    this.handleUpdate(data);
+
+    if (step !== this.activeStep) {
+      return;
+    }
+
+    const {orgId} = this.props.params;
+    const nextStep = ONBOARDING_STEPS[this.activeStepIndex + 1];
+
+    recordAnalyticStepComplete({
+      organization: this.props.organization,
+      project: this.firstProject,
+      step: nextStep,
+    });
+
+    browserHistory.push(`/onboarding/${orgId}/${nextStep.id}/`);
+  }
+
+  handleReturnToStep(step, data) {
+    const {orgId} = this.props.params;
+
+    this.handleUpdate(data);
+    browserHistory.push(`/onboarding/${orgId}/${step.id}/`);
+  }
+
+  scrollToActiveStep = () => {
+    const step = this.activeStep;
+    scrollToElemennt(`#onboarding_step_${step.id}`, {
+      align: 'middle',
+      duration: 300,
+    });
+  };
+
+  renderProgressBar() {
+    const activeStepIndex = this.activeStepIndex;
+    return (
+      <ProgressBar>
+        {ONBOARDING_STEPS.map((step, index) => (
+          <ProgressStep active={activeStepIndex === index} key={step.id} />
+        ))}
+      </ProgressBar>
+    );
+  }
+
+  renderOnboardingSteps() {
+    const activeStepIndex = this.activeStepIndex;
+    const {orgId} = this.props.params;
+
+    if (activeStepIndex === -1) {
+      // XXX(epurkhiser): Not super great to have a side effect during render,
+      // but I can live with it for now.
+      browserHistory.push(`/onboarding/${orgId}/welcome/`);
```

Comment:
Is it not possible to do this during componentDidUpdate?

Reply:
Yeah I can move it up there.

Reply:
4ff0d87